app/models/operatingsystems/coreos.rb

```diff
@@ -5,20 +5,26 @@ def pxe_type
     'coreos'
   end
 
-  def mediumpath(host)
-    medium_uri(host, "#{host.medium.path}/#{host.architecture.name}-usr").to_s.gsub('x86_64', 'amd64')
+  def mediumpath(medium_provider)
+    medium_provider.medium_uri("/$arch").to_s do |vars|
+      transform_vars(vars)
+    end
   end
 
   def url_for_boot(file)
     PXEFILES[file]
   end
 
   def pxedir
-    '$arch/$version'
+    '$arch-usr/$version'
```

Comment:
What's behind this change?

Reply:
It comes instead of `"#{host.medium.path}/#{host.architecture.name}-usr"` in mediumpath. Now it is more consistent.